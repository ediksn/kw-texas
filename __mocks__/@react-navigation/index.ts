jest.mock('@react-navigation/native', () => ({
  createNavigatorFactory: jest.fn(),
  useNavigation: () => ({
    navigate: jest.fn(),
    dispatch: jest.fn(),
    goBack: jest.fn(),
    setOptions: jest.fn()
  }),
  useRoute: jest.fn(),
  useFocusEffect: jest.fn()
}))

jest.mock('@react-navigation/material-top-tabs', () => ({
  createMaterialTopTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/stack', () => ({
  createStackNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

jest.mock('@react-navigation/bottom-tabs', () => ({
  createBottomTabNavigator: () => ({
    Navigator: jest.mock,
    Screen: jest.mock
  })
}))

export {}
