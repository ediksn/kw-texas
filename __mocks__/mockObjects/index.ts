import { PostInterface } from '~/interfaces/postInterface'
import { ProfileDetail } from '~/interfaces/usrInterface'
import { CommentInterface } from '~/interfaces/commentInterface'
import { GroupInterface } from '~/interfaces/groupInterface'

export const postMockObject: PostInterface = {
  bookmarksCount: 0,
  content:
    '{"blocks":[{"key":"b4fr9","text":"Post with images","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
  createdAt: '2021-11-26T14:44:17.187Z',
  creatorId: '60ca46ca877e063418825130',
  creatorLastName: 'Murphy',
  creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
  creatorfirstName: 'Irene',
  deleted: false,
  detail: {
    images: [
      {
        url: 'some url',
        id: 'saddsas'
      }
    ]
  },
  groupInfo: { id: '60f5c7224ad3ae0019a91973', name: 'Trending Group' },
  id: '61a0f2c1f48c2600181c8add',
  isUserCreatorOfThePost: true,
  likedBy: [],
  likesCount: 0,
  pinned: false,
  repliesCount: 0,
  type: 'FILE',
  userHasAlreadyBookmarked: false,
  userHasAlreadyLiked: false
}

export const profileDetailMock: ProfileDetail = {
  kw_uid: 0,
  name: 'Test',
  username: 'Test',
  first_name: 'Test',
  last_name: 'Test',
  photo_url: '',
  phone: '00000000',
  email: 'test',
  location: { city: 'test', state: 'test' },
  roles: ['Test', 'test'],
  following: false,
  mobile_phone: '0000000',
  service_area: 'test;test',
  bio: 'bio test',
  banner: 'test',
  closed_volume: '',
  closed_units: '',
  contracts_written_u: '',
  contracts_written_volume: '',
  listings_taken_u: '',
  listings_taken_dl: '',
  listings_sold_u: '',
  listings_sold_dl: '',
  listings_written_u: '',
  listings_written_volume: '',
  listings_closed_u: '',
  listings_closed_volume: '',
  buy_sales_closed_u: '',
  buy_sales_closed_volume: '',
  lease_written_u: '',
  lease_written_volume: '',
  lease_closed_u: '',
  lease_closed_volume: '',
  refer_u: '',
  refer_volume: '',
  average_sale_price: '',
  kw_start_date: '',
  languages_spoken: '',
  facebook: '',
  twitter: '',
  youtube: '',
  linkedin: '',
  google_plus: '',
  instagram: '',
  preferred_communication_method: '',
  time_zone: '',
  referral_opt_in: true,
  channels: [],
  profile_display_options: {
    memberships: false,
    languages: false,
    closed_units: false,
    listings_sold_units: false,
    buy_sales_closed_units: false,
    lease_closed_units: false,
    time_with_kw: false
  }
}
export const commentMockObject: CommentInterface = {
  id: '61a0f2c1f48c2600181c8ad2',
  creatorId: '61a0f31f48c2600181c8add',
  createdAt: '2021-11-26T14:44:17.187Z',
  type: 'STANDARD',
  creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
  creatorLastName: 'Murphy',
  creatorFirstName: 'Irene',
  deleted: false,
  postId: '61a0f2c1f48c2600181c8add',
  updatedAt: '2021-11-26T14:44:17.187Z',
  detail: {
    content: 'test comment',
    edited: false,
    likesCount: 0,
    userHasAlreadyLiked: false,
    media: null
  }
}

export const groupMockObject: GroupInterface = {
  id: '1',
  description: 'description',
  icon: {
    filename: 'hello.jpg',
    id: '1',
    url: 'https://avatar.kwconnect.com/556397.jpeg'
  },
  membersCount: 1,
  name: 'Test',
  postCount: 1,
  status: 1
}
