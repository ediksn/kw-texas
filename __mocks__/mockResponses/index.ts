import moment from 'moment'
import { PostInterface } from '~/interfaces/postInterface'

export const videoResponse = {
  id: 2230,
  title: 'zzz',
  description: 'zzz',
  videoUrl: 'file:///data/user/0/com.connectmobile/cache/Camera/b52e283c-40a5-40d0-9cf9-9a51843ec654.mp4',
  imageUrl: null,
  status: 'FINISHED',
  privacy: 'PUBLIC',
  publishedBy: null,
  liked: false,
  createdAt: '1630700559000',
  likesDetail: { likes: 0, __typename: 'Like' },
  bookmarked: false,
  timeRecorded: 1,
  views: 2,
  agents: [
    {
      id: 556397,
      kwUid: 556397,
      firstName: 'Irene',
      lastName: 'Murphy',
      __typename: 'Agent'
    }
  ],
  creatorHashtags: [],
  scripts: [
    {
      videoUrl: 'https://storage.googleapis.com/script-hub-practice-prod-bc0f/5_Waive_the_Appraisal_OBJ_1_low.mp4',
      videoResponseUrl:
        'https://storage.googleapis.com/script-hub-practice-prod-bc0f/5_Waive_the_Appraisal_RESPONSE_1_low.mp4',
      shortDescription:
        'I am hearing that people are waiving their appraisals and I do not want to do that, what does that even mean?',
      hashtags: [
        {
          id: 9,
          description: 'Appointment',
          tag: 'Appointment',
          __typename: 'Hashtag'
        }
      ],
      topics: [
        {
          id: 1,
          title: 'Lead Generation',
          __typename: 'Topic'
        }
      ],
      __typename: 'Script'
    }
  ],
  formats: [],
  __typename: 'ScriptMeeting'
}

export const profileResponse = {
  id: 532768,
  assigneeId: '556397',
  name: 'Kelle Test Team',
  sortName: 'Kelle Test Team',
  role: 'Buyer Specialist',
  photo: 'https://storage.googleapis.com/attachment-qa-eeac/556398/c5dd00kacb05iv34efp0.jpg',
  isPersonal: false,
  isDefault: true,
  members: [28947, 183149],
  userProfile: {
    photo: 'https://avatar.kwconnect.com/556397.jpeg',
    kw_uid: 556397,
    username: 'kelle2',
    first_name: 'Irene',
    last_name: 'Murphy',
    kw_email: null,
    default_org_id: 532768,
    recovery_email: 'kelle2@kw.com',
    email: 'kelle2@kw.com',
    marketing_email: 'tssv@kw.com',
    kw_start_dt: moment('2019-01-14'),
    re_start_dt: null,
    phone: '512-327-3070',
    mobile_phone: null,
    license: null,
    address_1: '1221 South Mopac Expy',
    address_2: 'Ste 400',
    city: 'Austin',
    country: 'US',
    state: 'TX',
    postal_code: '78746'
  }
}

export const postResponse: PostInterface = {
  id: '6172c50ac8a0fb001c4ddf0a',
  content: 'Great2',
  type: 'FILE',
  likedBy: [],
  likesCount: 0,
  repliesCount: 0,
  bookmarksCount: 0,
  isUserCreatorOfThePost: true,
  creatorId: '60ca46ca877e063418825130',
  creatorfirstName: 'Irene',
  creatorLastName: 'Murphy',
  creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
  createdAt: '2021-10-22T14:04:58.171Z',
  deleted: false,
  pinned: false,
  userHasAlreadyLiked: false,
  userHasAlreadyBookmarked: false,
  groupInfo: {
    id: '60f603104d8d7a001ac9ec79',
    name: 'Aanother Best Test Group 2021-08-30T07:11.000Z'
  },
  detail: {
    links: [
      {
        url: 'https://www.youtube.com/channel/UCDlujZcssCwV4P9XItrhC0A'
      }
    ]
  }
}

export const postsResponse = {
  data: {
    data: {
      getPosts: [
        {
          id: '6193d00ade67e4001bbba16e',
          content:
            '{"blocks":[{"key":"1l4c3","text":"J","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-16T15:36:42.438Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61923b6cee9a52001cc268ec',
          content:
            '{"blocks":[{"key":"8kmh4","text":"Now posr ","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-15T10:50:20.180Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '610a9219d330e7001a0c8627', name: 'test_closed' }
        },
        {
          id: '618c0d7912357f001a886ba3',
          content:
            '{"blocks":[{"key":"dovh7","text":"Check out these new homes!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'FILE',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:20:41.473Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9b', name: 'Demo Market Center' }
        },
        {
          id: '618c0ce1d55ede00191adaab',
          content:
            '{"blocks":[{"key":"ehu3r","text":"Hey, Demo Market Center!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:18:09.317Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9b', name: 'Demo Market Center' }
        },
        {
          id: '618c0b51d55ede00191adaa9',
          content:
            '{"blocks":[{"key":"9892l","text":"Hey, this is a post!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: ['60ca46ca877e063418825130'],
          likesCount: 1,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-10T18:11:29.901Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9a', name: 'Keller Williams Realty International' }
        },
        {
          id: '618af740d55ede00191ada36',
          content:
            '{"blocks":[{"key":"q97e","text":"I\'m sharing something!","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-09T22:33:36.127Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '61672a0c51c4f9001b9bec9a', name: 'Keller Williams Realty International' }
        },
        {
          id: '618978615de222001c15c163',
          content:
            '{"blocks":[{"key":"2r209","text":"https://www.youtube.com/watch?v=aV2Dfcz20xQ","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[{"offset":0,"length":43,"key":0}],"data":{}}],"entityMap":{"0":{"type":"LINK","mutability":"MUTABLE","data":{"href":"https://www.youtube.com/watch?v=aV2Dfcz20xQ","rel":"noopener noreferrer","target":"_blank","url":"https://www.youtube.com/watch?v=aV2Dfcz20xQ"}}}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T19:20:01.313Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '618973876c44c0001e5ba233',
          content:
            '{"blocks":[{"key":"bjpoc","text":"https://www.youtube.com/watch?v=daE2-3T5iiU","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T18:59:19.164Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61896dc55de222001c15c161',
          content:
            '{"blocks":[{"key":"eeaik","text":"Hello Irene Murphy ! What\'s going on? #kw","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[{"offset":6,"length":12,"key":0}],"data":{}}],"entityMap":{"0":{"type":"mention","mutability":"SEGMENTED","data":{"mention":{"name":"Irene Murphy","avatar":"https://avatar.kwconnect.com/556397.jpeg","link":"https://avatar.kwconnect.com/556397.jpeg"}}}}}',
          type: 'STANDARD',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60f5b6cbc08afb001d3ba9bc',
          isUserCreatorOfThePost: true,
          creatorfirstName: 'Michael',
          creatorLastName: 'Reid',
          creatorPhoto: 'https://avatar.kwconnect.com/556398.jpeg',
          createdAt: '2021-11-08T18:34:45.856Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '60f603104d8d7a001ac9ec79', name: 'Aanother Best Test Group 2021-08-30T07:11.000Z' }
        },
        {
          id: '61854d2a77f135001cee788b',
          content:
            '{"blocks":[{"key":"99lm1","text":"r","type":"unstyled","depth":0,"inlineStyleRanges":[],"entityRanges":[],"data":{}}],"entityMap":{}}',
          type: 'FILE',
          likedBy: [],
          likesCount: 0,
          repliesCount: 0,
          bookmarksCount: 0,
          creatorId: '60ca46ca877e063418825130',
          isUserCreatorOfThePost: false,
          creatorfirstName: 'Irene',
          creatorLastName: 'Murphy',
          creatorPhoto: 'https://avatar.kwconnect.com/556397.jpeg',
          createdAt: '2021-11-05T15:26:34.237Z',
          deleted: false,
          pinned: false,
          userHasAlreadyLiked: false,
          userHasAlreadyBookmarked: false,
          groupInfo: { id: '610a9fc5d330e7001a0c868b', name: 'test_public_with_members' }
        }
      ]
    }
  }
}

export const createPostResponse = {
  data: {
    createPost: '6172c50ac8a0fb001c4ddf0a'
  }
}

export const addCommentToPostResponse = {
  data: {
    replyPost: '61b899362f3ef4001a768d59'
  }
}

export const eventResponse = {
  bookmarked: true,
  ends: moment('2021-10-11 19:00:00'),
  id: 857429,
  instructors: [
    {
      firstName: 'Gabi',
      lastName: 'Holmberg'
    }
  ],
  location: 'Online',
  name: 'Online Event for KW Texas Agents',
  price: '0',
  published: true,
  slug: null,
  starts: moment('2021-10-10 11:00:00')
}

export const eventsResponse = {
  data: {
    searchOnlineEvents: [
      {
        id: 857784,
        name: 'AT By API_2021-11-01T07:08:31.613',
        location: 'Online',
        starts: '2021-11-01 08:08:31',
        ends: '2021-11-03 11:18:31',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857788,
        name: 'AT By API_2021-11-01T07:08:36.352',
        location: 'Online',
        starts: '2021-11-01 08:08:36',
        ends: '2021-11-03 11:18:36',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      },
      {
        id: 857789,
        name: 'AT By API_2021-11-01T07:08:36.734',
        location: 'Online',
        starts: '2021-11-01 08:08:36',
        ends: '2021-11-03 11:18:36',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857790,
        name: 'AT By API_2021-11-01T07:08:38.294',
        location: 'Online',
        starts: '2021-11-01 08:08:38',
        ends: '2021-11-03 11:18:38',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      },
      {
        id: 857791,
        name: 'AT By API_2021-11-01T07:08:40.191',
        location: 'Online',
        starts: '2021-11-01 08:08:40',
        ends: '2021-11-03 11:18:40',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857792,
        name: 'AT By API_2021-11-01T07:08:41.628',
        location: 'Online',
        starts: '2021-11-01 08:08:41',
        ends: '2021-11-03 11:18:41',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857488,
        name: 'THAT IS Test5555',
        location: 'Online',
        starts: '2021-11-01 11:00:00',
        ends: '2021-11-01 18:00:00',
        slug: null,
        price: '90',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857277,
        name: 'Recurring event for daily',
        location: 'Online',
        starts: '2021-11-01 23:00:00',
        ends: '2021-11-02 00:00:00',
        slug: null,
        price: '0',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857793,
        name: 'AT By API_2021-11-02T07:08:33.065',
        location: 'Online',
        starts: '2021-11-02 08:08:33',
        ends: '2021-11-04 11:18:33',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: false,
        published: true
      },
      {
        id: 857796,
        name: 'AT By API_2021-11-02T07:08:38.056',
        location: 'Online',
        starts: '2021-11-02 08:08:38',
        ends: '2021-11-04 11:18:38',
        slug: null,
        price: '123.12',
        instructors: [
          {
            firstName: 'Chris',
            lastName: 'Mondoux'
          }
        ],
        bookmarked: true,
        published: true
      }
    ]
  }
}
export const deletePostResponse = {
  data: { deletePost: true }
}

export const addBookmarkPostResponse = {
  data: { toggleBookmarkPost: true }
}

export const addLikePostResponse = {
  data: { toggleLikePost: true }
}

export const loginResponse = {
  id_token: 'id_token'
}

export const idTokenJWT = {
  at_hash: 'at_hash',
  sub: 'sub',
  kwoid: 'kwoid',
  auditTrackingId: 'auditTrackingId',
  iss: 'iss',
  tokenName: 'id_token',
  preferred_username: 'preferred_username',
  given_name: 'given_name',
  aud: 'aud',
  azp: 'azp',
  auth_time: 1634927734,
  name: 'name',
  realm: 'realm',
  exp: 1634931334,
  tokenType: 'tokenType',
  family_name: 'family_name',
  iat: 1634927734,
  email: 'email@kw.com',
  kwuid: 'kwuid'
}

export const pickerResponse = [
  {
    type: 'image/jpg',
    fileName: 'image1.jpg',
    width: 3000,
    height: 2002,
    uri: 'http://placehold.it/120x120&text=image1',
    fileSize: 6581432
  },
  {
    type: 'image/jpg',
    fileName: 'image2.jpg',
    width: 3000,
    height: 2002,
    uri: 'http://placehold.it/120x120&text=image2',
    fileSize: 6581432
  }
]
