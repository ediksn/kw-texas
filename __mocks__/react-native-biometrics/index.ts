jest.mock('react-native-biometrics', () => ({
  isSensorAvailable: jest.fn(() => Promise.resolve({ then: jest.fn() }))
}))

export {}
