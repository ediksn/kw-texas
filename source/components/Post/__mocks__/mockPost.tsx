const mockPost = {
  id: 1,
  author: {
    name: 'Nina Royals',
    picture: '__mocks__/avatarNinaRoyals.png'
  },
  content:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean tellus leo, fringilla at elit sed, tincidunt viverra dui. Nullam dignissim mi et tincidunt luctus. Nullam ut eleifend diam. Phasellus faucibus aliquam leo, vitae scelerisque nunc dignissim in. Maecenas vulputate quam id metus efficitur tristique. Ut maximus varius enim eleifend feugiat. Ut at ultricies lorem, eget molestie risus. Nam ultricies tempus sem quis fermentum. Sed a eros ac metus scelerisque hendrerit. Ut nec urna aliquam, sagittis eros non, bibendum risus. Ut congue eleifend velit eu dictum. Praesent eu mi vitae leo pulvinar ullamcorper. Sed tortor felis, tristique vitae lacinia sit amet, pellentesque ut felis. Nulla finibus erat libero, eu sodales metus auctor at. Duis iaculis leo vitae odio mollis, feugiat aliquam erat eleifend. Mauris quis dui at lorem tincidunt viverra nec vel mauris. Pellentesque laoreet dapibus purus, eget vehicula nunc scelerisque quis. Donec ac nibh sed erat consequat sagittis ut ut odio. Curabitur tempor vitae neque vel gravida. Morbi varius ligula nec tellus accumsan, quis pharetra ante dictum. Nulla pharetra ac ex id cursus. Vivamus ut lacinia nunc. Maecenas nec orci risus.',
  date: '4 hours ago',
  likes: 13,
  comments: 5,
  shares: 2
}

export default mockPost
