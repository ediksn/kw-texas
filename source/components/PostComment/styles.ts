import { StyleSheet } from 'react-native'
import { verticalScale, scale } from 'react-native-size-matters'
import { theme } from '~/constants'

export const styles = StyleSheet.create({
  commentWrapper: {
    flexDirection: 'row',
    justifyContent: 'flex-start',
    paddingHorizontal: scale(12),
    paddingVertical: verticalScale(4)
  },
  commentAvatar: {
    paddingRight: scale(10)
  },
  commentContent: {
    flex: 1
  },
  commentBody: {
    paddingVertical: verticalScale(10),
    paddingHorizontal: scale(10),
    borderRadius: scale(4),
    backgroundColor: theme.backgrounds.greyBackground
  },
  commentAuthor: {
    paddingBottom: scale(10),
    fontSize: theme.fonts.SMALL_SIZE,
    fontFamily: 'Mulish-Bold',
    textTransform: 'uppercase',
    color: theme.post.name
  },
  commentText: {
    color: theme.post.content
  },
  commentFooter: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingTop: verticalScale(5)
  },
  commentDate: {
    color: theme.greenColor,
    fontSize: theme.fonts.SMALL_SIZE
  },
  commentActions: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  commentLike: {
    color: theme.greenColor,
    fontSize: theme.fonts.SMALL_SIZE,
    paddingHorizontal: scale(4)
  },
  commentComment: {
    color: theme.greenColor,
    fontSize: theme.fonts.SMALL_SIZE,
    paddingHorizontal: scale(4)
  },
  showMore: {
    fontSize: theme.fonts.SMALL_SIZE,
    fontFamily: 'Mulish-Bold',
    color: theme.post.green
  }
})
