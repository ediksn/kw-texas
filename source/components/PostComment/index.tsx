import React, { useCallback, useEffect, useState } from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import moment from 'moment'
import { styles } from './styles'
import Avatar from '~/components/Avatar'
import { CommentInterface } from '~/interfaces/commentInterface'
import { useUnRichContent } from '~/hooks'

const PostComment = ({ item }: { item: CommentInterface }) => {
  const {
    createdAt,
    creatorFirstName,
    creatorLastName,
    creatorPhoto,
    detail: { content }
  } = item
  const date = moment(createdAt).format('MM/DD/YY')
  const [numberOfLines, setNumberOfLines] = useState<number | undefined>()
  const [hasShowLessMore, setHasShowLessMore] = useState(false)
  const [showMore, setShowMore] = useState(false)
  const MAX_LINES = 5
  const contentText = useUnRichContent(content)

  const addMoreText = useCallback(
    (event: any) => {
      if (event.nativeEvent.lines.length > MAX_LINES && !showMore) {
        setHasShowLessMore(true)
      }
    },
    [showMore]
  )

  useEffect(() => {
    if (hasShowLessMore) {
      setNumberOfLines(showMore ? undefined : MAX_LINES)
    }
  }, [hasShowLessMore, showMore])

  return (
    <View style={styles.commentWrapper}>
      <View style={styles.commentAvatar}>
        <Avatar uri={creatorPhoto} />
      </View>
      <View style={styles.commentContent}>
        <View style={styles.commentBody}>
          <Text style={styles.commentAuthor}>{`${creatorFirstName} ${creatorLastName}`}</Text>
          <Text
            style={styles.commentText}
            numberOfLines={numberOfLines}
            onTextLayout={addMoreText}
            ellipsizeMode='clip'
          >
            {contentText}
          </Text>
          {hasShowLessMore && (
            <TouchableOpacity onPress={() => setShowMore(!showMore)}>
              <Text style={styles.showMore}>Show {showMore ? 'less' : 'more'}</Text>
            </TouchableOpacity>
          )}
        </View>
        <View style={styles.commentFooter}>
          <Text style={styles.commentDate}>{date}</Text>
          <View style={styles.commentActions}>
            <Text style={styles.commentLike}>Like</Text>
            <Text style={styles.commentComment}>Comment</Text>
          </View>
        </View>
      </View>
    </View>
  )
}

export default PostComment
