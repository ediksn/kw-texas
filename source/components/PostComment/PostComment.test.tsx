import React from 'react'
import { render } from '@testing-library/react-native'
import PostComment from '~/components/PostComment/index'
import { commentMockObject } from '../../../__mocks__/mockObjects'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

jest.mock('react-redux', () => ({
  useSelector: jest.fn()
}))

jest.mock(
  'rn-fetch-blob',
  () => {
    return {
      DocumentDir: () => {},
      ImageCache: {
        get: {
          clear: () => {}
        }
      },
      fs: {
        exists: jest.fn(),
        dirs: {
          MainBundleDir: () => {},
          CacheDir: () => {},
          DocumentDir: () => {}
        }
      }
    }
  },
  { virtual: true }
)

test('renders correctly', () => {
  const header = render(<PostComment item={commentMockObject} />).toJSON()
  expect(header).toMatchSnapshot()
})
