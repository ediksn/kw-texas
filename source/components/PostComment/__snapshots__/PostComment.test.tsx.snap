// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "paddingHorizontal": 25.714285714285715,
      "paddingVertical": 7.847058823529411,
    }
  }
>
  <View
    style={
      Object {
        "paddingRight": 21.428571428571427,
      }
    }
  >
    <Image
      onError={[Function]}
      resizeMode="cover"
      source={
        Object {
          "uri": "https://avatar.kwconnect.com/556397.jpeg",
        }
      }
      style={
        Array [
          Object {
            "borderRadius": 31.428571428571427,
            "height": 62.857142857142854,
            "width": 62.857142857142854,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#E4E5E7",
          "borderRadius": 8.571428571428571,
          "paddingHorizontal": 21.428571428571427,
          "paddingVertical": 19.61764705882353,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#282B33",
            "fontFamily": "Mulish-Bold",
            "fontSize": 18.857142857142858,
            "paddingBottom": 21.428571428571427,
            "textTransform": "uppercase",
          }
        }
      >
        Irene Murphy
      </Text>
      <Text
        ellipsizeMode="clip"
        onTextLayout={[Function]}
        style={
          Object {
            "color": "#3D424D",
          }
        }
      >
        test comment
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingTop": 9.808823529411764,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#6C727E",
            "fontSize": 18.857142857142858,
          }
        }
      >
        11/26/21
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#6C727E",
              "fontSize": 18.857142857142858,
              "paddingHorizontal": 8.571428571428571,
            }
          }
        >
          Like
        </Text>
        <Text
          style={
            Object {
              "color": "#6C727E",
              "fontSize": 18.857142857142858,
              "paddingHorizontal": 8.571428571428571,
            }
          }
        >
          Comment
        </Text>
      </View>
    </View>
  </View>
</View>
`;
